org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/HoverHandlerTest.java

```diff
@@ -342,6 +342,83 @@ public void testHoverWithAttachedJavadoc() throws Exception {
 		assertTrue("Unexpected hover :\n" + content, content.contains("**Author:**"));
 	}
 
+	@Test
+	public void testHoverOnJavadocWithValueTag() throws Exception {
+		importProjects("maven/salut");
+		project = WorkspaceHelper.getProject("salut");
+		handler = new HoverHandler(preferenceManager);
+		//given
+		String payload = createHoverRequest("src/main/java/java/Foo2.java", 12, 30);
+		TextDocumentPositionParams position = getParams(payload);
+
+		// when
+		Hover hover = handler.hover(position, monitor);
+		assertNotNull("Hover is null", hover);
+		assertEquals("Unexpected hover contents:\n" + hover.getContents(), 2, hover.getContents().size());
+		Either<String, MarkedString> javadoc = hover.getContents().get(1);
+		String content = null;
+		assertTrue("javadoc has null content", javadoc != null && javadoc.getLeft() != null && (content = javadoc.getLeft()) != null);
```

Comment:
please assert the actual content has the expected value

Reply:
Adding something like:
```java
	protected void assertMatches(String pattern, String value) {
		assertTrue(value + " doesn't match pattern: " + pattern, Pattern.matches(pattern, value));
	}
```
to AbstractProjectsManagerBasedTest, you can then assert the content matches some regexp, like:

assertMatches("\\[\"SimpleStringData\"\\]\\(file:/.*/salut/src/main/java/java/Foo2.java#13\\) is a simple String", content);